src/chronify/store.py

```diff
@@ -166,34 +261,62 @@ def ingest_table(
                 *[Column(x, y) for x, y in zip(df.columns, dtypes)],
             )
             table.create(self._engine)
+        else:
+            table_existed = True
 
         with self._engine.begin() as conn:
             write_database(df, conn, dst_schema.name, dst_schema.time_config)
             try:
                 check_timestamps(conn, table, dst_schema)
             except Exception:
                 conn.rollback()
-                if not table_exists:
+                if not table_existed:
                     table.drop(self._engine)
                     self.update_table_schema()
                 raise
+            if not table_existed:
+                self._add_schema(conn, dst_schema)
             conn.commit()
-        self.update_table_schema()
+
+        if not table_existed:
+            self.update_table_schema()
+
+    def list_tables(self) -> list[str]:
+        """Return a list of user tables in the database."""
+        return [x for x in self._metadata.tables if x != self.SCHEMAS_TABLE]
 
     def map_time(self, src_schema: TableSchema, dst_schema: TableSchema) -> None:
         """Map the existing table represented by src_schema to a new table represented by
         dst_schema.
         """
         map_time(self._engine, self._metadata, src_schema, dst_schema)
+        with self._engine.connect() as conn:
+            self._add_schema(conn, dst_schema)
+            conn.commit()
+        self.update_table_schema()
```

Comment:
I already called this in the mapper, probably okay to call it multiple times right?

Reply:
I'll remove it.